### ISSUE TITLE
too-many-instance-attributes counts InitVars

### ISSUE THREAD

**Issue by p4l1ly** [PyCQA/pylint#3794]
### Steps to reproduce
1. create a file `example.py`
```
"Module docstring"

from dataclasses import dataclass, InitVar

@dataclass
class Hello:
    "Class docstring"

    a_1: int
    a_2: int
    a_3: int
    a_4: int
    a_5: int
    a_6: int
    a_7: int
    a_8: InitVar[int]

    def __post_init__(self, a_8):
        self.a_7 += a_8
```
2. Run `pylint --reports no --persistent no example`

### Current behavior

**too-many-instance-attributes (8/7)** are reported but `a_8` is not an instance attribute so it should not be counted.

### Expected behavior

pylint should report no error.

### pylint --version output

```
pylint 2.6.0
astroid 2.4.2
Python 3.8.2 (default, Apr  8 2020, 14:31:25)
[GCC 9.3.0]
```
